homeassistant/components/evohome.py

```diff
@@ -0,0 +1,1941 @@
+# location_idx: 0        # if you have more than 1 location, use this
+
+These config parameters are YMMV:
+
+# use_heuristics: false  # this is for the highly adventurous person, YMMV
+# use_schedules: false   # this is for the slightly adventurous person
+# away_temp: 15.0        # °C, if you have a non-default Away temp
+# off_temp: 5.0          # °C, if you have a non-default Heating Off temp
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/evohome/
+"""
+
+# Glossary
+# TCS - temperature control system (a.k.a. Controller, Master), which can
+# have up to 13 Slaves:
+#   0-12 Heating zones (a.k.a. Zone), and
+#   0-1 DHW controller, (a.k.a. Boiler)
+
+# List of future features
+# Replace AutoWithEco: mode that allows a delta of +/-0.5, +/-1.0, +/-1.5, etc.
+# Improve hueristics: detect TRV Off, and OpenWindow
+
+import logging
+from datetime import datetime, timedelta
+import requests
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    ClimateDevice,
+
+    SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_OPERATION_MODE,
+    SUPPORT_AWAY_MODE,
+    SUPPORT_ON_OFF,
+
+    ATTR_CURRENT_TEMPERATURE,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_TEMP,
+    ATTR_TARGET_TEMP_STEP,
+    ATTR_OPERATION_MODE,
+    ATTR_OPERATION_LIST,
+    ATTR_AWAY_MODE,
+)
+
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    CONF_USERNAME,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+
+    EVENT_HOMEASSISTANT_START,
+    PRECISION_WHOLE,
+    PRECISION_HALVES,
+    PRECISION_TENTHS,
+
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNKNOWN,
+    TEMP_CELSIUS,
+
+    HTTP_BAD_REQUEST,
+    HTTP_TOO_MANY_REQUESTS,
+    HTTP_SERVICE_UNAVAILABLE,
+)
+
+# These are HTTP codes commonly seen with this component
+#   HTTP_BAD_REQUEST = 400          # usually, bad user credentials
+#   HTTP_TOO_MANY_REQUESTS = 429    # usually, api limit exceeded
+#   HTTP_SERVICE_UNAVAILABLE = 503  # this is common with Honeywell's websites
+
+from homeassistant.core import callback
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.temperature import display_temp as show_temp
+
+# these are specific to this component
+ATTR_UNTIL = 'until'
+
+# only the controller does client api I/O during update() to get current state
+# however, any entity can call methdos that will change state
+PARALLEL_UPDATES = 0
+
+# these are specific to this component
+CONF_HIGH_PRECISION = 'high_precision'
+CONF_USE_HEURISTICS = 'use_heuristics'
+CONF_USE_SCHEDULES = 'use_schedules'
+CONF_LOCATION_IDX = 'location_idx'
+CONF_AWAY_TEMP = 'away_temp'
+CONF_OFF_TEMP = 'off_temp'
+
+API_VER = '0.2.7'  # alternatively, '0.2.5' is the version used elsewhere in HA
+
+if API_VER == '0.2.7':  # these vars for >=0.2.6 (is it v3 of the api?)...
```

Comment:
This is very weird and should be removed. It should just be 0.2.7.

Reply:
This will be taken out.